dit/dit.py

```diff
@@ -327,90 +359,73 @@ def _export_all(self, concluded, verbose):
+        logbook = data.get('logbook', [])
+        if len(logbook) > 0:
+            last = logbook[-1]
             if not last['out']:
                 print("Already clocked in")
                 return
-
-        data['logbook'] += [{'in': self._now(), 'out': None}]
+        data['logbook'] = logbook + [{'in': _now(), 'out': None}]
 
     def _clock_out(self, data):
-        if not data['logbook']:
+        logbook = data.get('logbook', [])
+        if len(logbook) == 0:
             print("Already clocked out")
             return
-        last = data['logbook'][-1]
+        last = logbook[-1]
         if last['out']:
             print("Already clocked out")
             return
-
-        last['out'] = self._now()
+        last['out'] = _now()
+        data['logbook'] = logbook
 
     def _conclude(self, data):
         if data.get('concluded_at', None):
             print("Already concluded")
             return
-        data['concluded_at'] = self._now()
+        data['concluded_at'] = _now()
 
     # ===========================================
     # Parsers
 
     def _gid_parse(self, argv):
         ids = argv.pop(0).split(self.separator)
-        ids = list(map(int, [i if i else "0" for i in ids]))
         if len(ids) > 3:
-            raise Exception("Invalid GID format")
+            raise Exception("Invalid gID format")
+        ids = [int(i) if i else "0" for i in ids]
         ids = ids + [None] * (3 - len(ids))
         group_id, subgroup_id, task_id = ids
 
         group = self.index[group_id][0]
 
-        if subgroup_id is None:
-            subgroup_id = 0
-            for i in range(len(self.index[group_id][1])):
-                if self.index[group_id][1][i][0] == self.current_subgroup:
-                    subgroup_id = i
-                    break
-
-        subgroup = self.index[group_id][1][subgroup_id][0]
+        subgroup = None
+        task = None
 
-        if task_id:
-            task = self.index[group_id][1][subgroup_id][1][task_id]
-        else:
-            task = None
+        if subgroup_id:
+            subgroup = self.index[group_id][1][subgroup_id][0]
+            if task_id:
+                task = self.index[group_id][1][subgroup_id][1][task_id]
 
         return (group, subgroup, task)
 
     def _id_parse(self, argv):
         ids = argv.pop(0).split(self.separator)
-        ids = list(map(int, [i if i else "0" for i in ids]))
         if len(ids) > 3:
-            raise Exception("Invalid GID format")
+            raise Exception("Invalid ID format")
+        ids = [int(i) if i else "0" for i in ids]
         ids = [None] * (3 - len(ids)) + ids
         group_id, subgroup_id, task_id = ids
 
         if group_id is None:
-            group_id = 0
             for i in range(len(self.index)):
                 if self.index[i][0] == self.current_group:
                     group_id = i
                     break
 
         if subgroup_id is None:
-            subgroup_id = 0
```

Comment:
This is intended. The code will fail `if (sub)group_id == None and current_(sub)group == None`. I expect some wrong index exception. I will create an issue for properly handling errors.